matter/si91x/siwx917/BRD4338A/autogen/sl_event_handler.c

```diff
@@ -27,35 +31,26 @@ void sl_platform_init(void)
   osKernelInitialize();
 }
 
-void sl_kernel_start(void)
-{
-  osKernelStart();
-}
+void sl_kernel_start(void) { osKernelStart(); }
 
-void sl_driver_init(void)
-{
+void sl_driver_init(void) {
   button_init_instances();
   led_init_instances();
 }
 
-void sl_service_init(void) 
-{
-#ifdef DISPLAY_ENABLED
+void sl_service_init(void) {
+#if SL_ICD_ENABLED
+  sl_si91x_power_manager_init();
+  sli_si91x_power_manager_configure_ram_and_peripheral();
+#endif // SL_ICD_ENABLED
+#ifdef DISPLAY_ENABLED || SL_ICD_ENABLED

```

Comment:
why is the sleeptimer timer tied with Display?

Reply:
A periodic sleeptimer is used for the periodic maintenance for the LCD. That's why, we are shutting down lcd since it is restarting a periodic timer.

Reply:
It's in the platform code, https://github.com/SiliconLabs/wiseconnect/blob/master/components/device/silabs/si91x/mcu/drivers/hardware_drivers/memlcd/src/sl_memlcd.c#L148-L171